File: 0129-Sum-Root-to-Leaf-Numbers/0129.py
class Solution:
    def sumNumbers(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        result = 0
        if not root:
            return result

        path = [(0, root)]
        while path:
            pre, node = path.pop()
            if node:
                if not node.left and not node.right :
                    result += pre*10 + node.val
                
                path += [(pre*10 + node.val, node.right), (pre*10 + node.val, node.left)]
            
        return result

# class Solution:
#     def _sumNumbers(self, root, value):
#         if root:
#             self._sumNumbers(root.left, value*10+root.val)
#             self._sumNumbers(root.right, value*10+root.val)
#             if not root.left and not root.right:
#                 self.result += value*10 + root.val

#     def sumNumbers(self, root):
#         """
#         :type root: TreeNode
#         :rtype: int
#         """
#         self.result = 0
#         self._sumNumbers(root, 0)
#         return self.result

File: src/0129-Sum-Root-to-Leaf-Numbers/0129.cpp
#include <iostream>
#include <string>
#include <vector>
#include <queue>
#include <algorithm>
using namespace std;

//  Definition for a binary tree node.
struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode(int x) : val(x), left(NULL), right(NULL) {}
};

static int x = []() {std::ios::sync_with_stdio(false); cin.tie(0); return 0; }();

class Solution 
{
public:
    int sumNumbers(TreeNode* root) 
    {
        int result = 0;
        _sumNumbers(root, 0, result);
        return result;
    }
private: 
    void _sumNumbers(TreeNode* root, int value, int& result)
    {
        if (root)
        {
            if (root->left == nullptr and root->right == nullptr)
                result += value*10 + root->val;
            _sumNumbers(root->left, value*10 + root->val, result);
            _sumNumbers(root->right, value*10 + root->val, result);
        }
    }
};